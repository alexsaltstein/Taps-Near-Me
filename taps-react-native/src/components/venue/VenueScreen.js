import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import BackButton from '../widgets/BackButton';
import ErrorPage from '../widgets/ErrorPage';
import Loading from '../widgets/Loading';

const VenueScreen = ({ navigation, route }) => {
  const { id } = route.params;

  const [error, setError] = React.useState(false);
  const [loading, setLoading] = React.useState(true);

  const getData = async () => {
    try{
    }catch {
      setError(true);
    }
  }

  React.useEffect(()=> {
    (async () => {
      setLoading(true);
      await getData();
      setLoading(false);
    })()
  }, []);
  
  return (
    <View style={{flex: 1}}>
      { loading ?
        <Loading navigation={navigation}/> :
        <View style={{flex: 1}}>
          {
            error ?
              <ErrorPage navigation={navigation} />
              :
              <View style={styles.container}>
                <BackButton navigation={navigation}/>
                <Text>hii</Text>
                <Text>Venue Screen</Text>
                <Text>id: {id}</Text>
              </View>
          }
        </View>
      }
    </View>
  )
}

const styles = StyleSheet.create({
  flex1: {
    flex: 1
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
})

export default VenueScreen;