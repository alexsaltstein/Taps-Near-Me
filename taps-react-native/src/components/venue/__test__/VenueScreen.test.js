import '../../../matchMedia.mock';
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import VenueScreen from '../VenueScreen';
import { RecoilRoot } from 'recoil';
import { render, fireEvent } from '@testing-library/react-native'
import { View, Text, StyleSheet, TouchableOpacity, TextInput } from 'react-native';
import ReactDOM from 'react-dom';

const mockNavigation = {
  navigate: jest.fn(),
  addListener: jest.fn()
}

const mockRoute = {
    params: jest.fn()
}

test('renders correctly', () => {
  const tree = renderer.create(<RecoilRoot><VenueScreen navigation= {mockNavigation} route= {mockRoute}/></RecoilRoot>).toJSON();
  expect(tree).toMatchSnapshot();
});