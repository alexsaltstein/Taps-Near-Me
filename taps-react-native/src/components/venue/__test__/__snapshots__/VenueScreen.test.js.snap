// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-14lw9ot r-flex-13awgt0 r-width-13qz1uu"
  >
    <div
      className="css-view-1dbjc4n r-alignSelf-k200y r-cursor-1loqt21 r-justifyContent-1777fci r-left-1t68eob r-position-u8s1d r-top-12pp1o0 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
      data-focusable={true}
      onClick={[Function]}
      onContextMenu={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "WebkitTransitionDuration": "0s",
          "transitionDuration": "0s",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
        style={
          Object {
            "height": "30px",
            "width": "30px",
          }
        }
      >
        <div
          className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
          style={
            Object {
              "WebkitFilter": "url(#tint-0)",
              "filter": "url(#tint-0)",
            }
          }
        />
        <svg
          style={
            Object {
              "height": 0,
              "position": "absolute",
              "visibility": "hidden",
              "width": 0,
            }
          }
        >
          <defs>
            <filter
              id="tint-0"
              suppressHydrationWarning={true}
            >
              <feFlood
                floodColor="#000000"
              />
              <feComposite
                in2="SourceAlpha"
                operator="atop"
              />
            </filter>
          </defs>
        </svg>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-backgroundColor-1gbixzw r-borderRadius-1ylenci r-justifyContent-1777fci r-marginTop-1as70bj r-paddingBottom-k8qxaj r-paddingRight-1qfoi16"
    >
      <div
        className="css-view-1dbjc4n r-height-7r4507 r-width-1iadsmv"
        style={Object {}}
      >
        <div
          className="css-view-1dbjc4n r-height-7r4507 r-width-1iadsmv"
          style={Object {}}
        />
      </div>
    </div>
  </div>
</div>
`;
